services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/controller/SegmentController.java

```diff
@@ -72,6 +73,16 @@ public SegmentWebResponseJson createSegment(@PathVariable String projectid, @Req
             RestPreconditionsUtil.verifyResourceNotNull(segmentWebResponseObject);
             RestPreconditionsUtil.populateResourceProjectId(segmentWebResponseObject, projectid);
 
+            // verify network type
+            String networkType = segmentWebRequestObject.getNetworkType();
+            if (networkType.equals(NetworkTypeEnum.VXLAN)) {
```

Comment:
Like switching to enum here. 

But what if networkType == null..

Reply:
I think  networkType == null condition could be checked when we verify resource of SegmentWebRequestJson. If found networkType == null, we could throw an specific exception.
So I miss networkType == null here.
What do you think?